xbmc/network/DNSNameCache.cpp

```diff
@@ -106,24 +112,39 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
 
 bool CDNSNameCache::GetCached(const std::string& strHostName, std::string& strIpAddress)
 {
-  CSingleLock lock(m_critical);
-
-  // loop through all DNSname entries and see if strHostName is cached
-  for (int i = 0; i < (int)g_DNSCache.m_vecDNSNames.size(); ++i)
   {
-    CDNSName& DNSname = g_DNSCache.m_vecDNSNames[i];
-    if ( DNSname.m_strHostName == strHostName )
+    CSingleLock lock(m_critical);
+
+    // loop through all DNSname entries and see if strHostName is cached
+    for (int i = 0; i < (int)g_DNSCache.m_vecDNSNames.size(); ++i)
     {
-      strIpAddress = DNSname.m_strIpAddress;
-      return true;
+      CDNSName& DNSname = g_DNSCache.m_vecDNSNames[i];
+      if (DNSname.m_strHostName == strHostName)
+      {
+        strIpAddress = DNSname.m_strIpAddress;
+        return true;
+      }
     }
```

Comment:
Whilst you are here, can i request you update this for loop to the following

```
    // loop through all DNSname entries and see if strHostName is cached
    for (auto DNSname : g_DNSCache.m_vecDNSNames)
    {
      if (DNSname.m_strHostName == strHostName)
      {
        strIpAddress = DNSname.m_strIpAddress;
        return true;
      }
    }
```

We are slowly trying to modernise things that get touched in PR's

Reply:
I noticed today that i stuffed up the above diff. Really should be 

```
    for (const auto& DNSname : g_DNSCache.m_vecDNSNames)
```

I can do a followup fix to save you having to redo it again

Reply:
Changed.

Reply:
Cheers, thanks for that.